src/Cardano/Wallet/Primitive/Signing.hs

```diff
@@ -65,37 +63,25 @@ mkStdTx
     -> [TxOut]
     -- ^ Selected outputs (including change)
     -> Either SignTxError (Tx, [TxWitness])
-mkStdTx s (rootPrv, pwd) ownedIns outs = do
+mkStdTx s creds@(_, pwd) ownedIns outs = do
     let ins = (fmap fst ownedIns)
         tx = Tx ins outs
         txSigData = hashTx tx
-
-    txWitnesses <- forM ownedIns (\(_in, TxOut addr _c) ->
-        mkWitness txSigData <$> keyFrom' addr)
-
+    txWitnesses <- forM ownedIns $ \(_in, TxOut addr _c) -> mkWitness txSigData
+        <$> withEither (KeyNotFoundForAddress addr) (keyFrom addr creds s)
     return (tx, txWitnesses)
-
   where
-    keyFrom' addr =
-        -- We are ignoring the new state/pool. We won't discover any new
-        -- addresses when submitting transactions.
-        case (fst $ keyFrom addr (rootPrv, pwd) s) of
-            Just key -> Right key
-            Nothing -> Left $ KeyNotFoundForAddress addr
-
+    withEither :: e -> Maybe a -> Either e a
+    withEither e = maybe (Left e) Right
     hashTx :: Tx -> Hash "tx"
     hashTx txSigData = Hash
-        $ convert
-        $ (hash @ByteString @Blake2b_256)
+        $ BA.convert
```

Comment:
Small change here, using qualified import for ambiguous function. The context is probably enough here, but, usually, for functions like this that exist in many modules, it's good to be specific (like `map` or `filter` or `encode` ... )